dali/python/nvidia/dali/pipeline.py

```diff
@@ -477,26 +493,48 @@ def feed_input(self, data_node, data, layout="", cuda_stream = None):
         from nvidia.dali.external_source import _check_data_batch
         _check_data_batch(data, self._batch_size, layout)
 
+        infer_stream = False
+        if cuda_stream is None:
+            infer_stream = True
+        if cuda_stream == -1:
+            cuda_stream = None
+        else:
+            cuda_stream = types._raw_cuda_stream(cuda_stream)
```

Comment:
Can we maybe use another name for cuda_stream that is passed further as the argument to this function and the one we use later have a bit different meanings with the None and -1.

Reply:
Maybe something like `stream_ptr`, and make the `_raw_cuda_stream` already return a `c_void_p` values?
You're packing it by hand in every invocation place.

Reply:
Or maybe it would be better to have some boolean for `SetExternalTensorInput` indicating if the stream should be generated internally?

Reply:
Unpacking it as a raw pointer may cause hard errors in case of the following or similar scenario:
```
stream = torch.cuda.Stream()
fn.external_source(src, cuda_stream = stream)
# stream reference is forgotten
```
If we unwrap immediately, we'll lose the reference to the stream and will be destroyed - our stream pointer will be invalid and can even be recycled by the driver upon next stream creation, where we'd coincidentally have a _different_ stream, but still invalid. I don't want our users to debug THAT kind of erros.

Reply:
> Or maybe it would be better to have some boolean for SetExternalTensorInput indicating if the stream should be generated internally?

I think None is fine.